### ISSUE TITLE
The color-contrast rule doesn't take the CSS filter property into account.

### ISSUE THREAD

**Issue by matuzo** [dequelabs/axe-core#3214]
Product: axe Extension (and probably all other products)

Expectation: axe-core should discover contrast issues caused by the filter property. `opacity: 0.1` and `filter: opacity(0.1)` are [technically the same](https://developer.mozilla.org/en-US/docs/Web/CSS/filter#opacity) and should be treated the same.

Actual: The color-contrast rule doesn't take the filter property into account.

Here's a demo: https://codepen.io/matuzo/debug/LYjpQPZ

<img width="1792" alt="Screenshot 2021-10-14 at 20 01 38" src="https://user-images.githubusercontent.com/1496761/137371660-f0edf85e-9e93-4d28-82a0-d8a1dc163889.png">

<pre><code>
axe-core version: 4.3.3
Chrome extension
Chrome 94
macOS 11.6
</code></pre>

**Comment by matuzo** [comment]
Thanks for the quick reply!

No, actually quite the opposite. I've revisited [an article I wrote 2,5 years ago](https://www.matuzo.at/blog/building-the-most-inaccessible-site-possible-with-a-perfect-lighthouse-score/) about automatic accessibility testing where I've created an very unrealistic scenario. I noticed that using `opacity: 0.3` triggered an error in axe which didn't happen 2,5 years ago, so I came up with the “filter solution”.

**Comment by WilcoFiers** [comment]
Thanks! Well we'll be sure to look into these too. Filter is really tough to support, but we'll figure something out.

**Comment by scurker** [comment]
@WilcoFiers `filter` with `contrast` can cause false positives:

```html
<span class="filter">hello world</span>
```

```css
.filter{
  filter: invert(1) contrast(2);
  background-color: #ccc;
  color: #676767;
}
```

**Comment by WilcoFiers** [comment]
@scurker Yeah, axe-core doesn't handle filter at all. Again, if you encounter this causing false positives in a real-world website, let me know. It's certainly something that I'd like to see us address, but real-world false positives get priority.
